include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -150,11 +153,12 @@ typename MatType::elem_type NSGA2::Optimize(
 
     // Perform crowding distance assignment.
     crowdingDistance.resize(population.size());
-
-    for (size_t fNum = 0; fNum < fronts.size(); fNum++)
+    std::fill(crowdingDistance.begin(), crowdingDistance.end(), 0.f);
```

Comment:
Again, a simple and effective 'for' loop was replaced with something more elaborate.  There is no need for this.

Reply:
Are you talking about the std::fill or std::for_each?

Reply:
Because I moved crowdingDistance zero intialisation out from the function. So instead of setting 0 to that vector per front, I did it at once. 

If however you're talking about std::for_each, I'll revert that change.